### PR TITLE
Get Full Retrieval Intent Name

### PR DESCRIPTION
**Pull request by vcidst**
**Proposed changes**:
- Adds `FULL_RETRIEVAL_INTENT_NAME_KEY` to the intent dictionary. This has downstream effects and fixes a bug in the Analytics pipeline that expects this key
- Works on [ATO-1956](https://rasahq.atlassian.net/browse/ATO-1956)

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

[ATO-1956]: https://rasahq.atlassian.net/browse/ATO-1956?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: rasa/core/processor.py [hunk 1]
```diff
@@ -732,6 +737,23 @@ async def parse_message(
 
         return parse_data
 
+    def _update_full_retrieval_intent(self, parse_data: Dict[Text, Any]) -> None:
```
Comment (ancalita) [inline rasa/core/processor.py:740]: Could we add a processor unit test for this case please?

Comment (vcidst) [reply rasa/core/processor.py:740]: Yep, just added it in the commit [acf376b](https://github.com/RasaHQ/rasa/pull/12998/commits/acf376b675d42acf8faaffff5564a2cad3e17ede)

**Review by ancalita — APPROVED** [review]
Nicely done 💯 Only the linting needs to be fixed.

File: tests/core/test_processor.py [hunk 2]
```diff
@@ -1928,3 +1933,62 @@ async def test_run_anonymization_pipeline_mocked_pipeline(
...
+
+
+async def test_update_full_retrieval_intent(
+    default_processor: MessageProcessor,
+) -> None:
+    parse_data = {
+        "text": "I like sunny days in berlin",
+        "intent": {"name": "chitchat", "confidence": 0.9},
+        "entities": [],
+        "response_selector": {
```
Comment (ancalita) [inline tests/core/test_processor.py:1945]: Not sure if this is a duplicate curly brace or meant to be a list of dicts instead?

Comment (vcidst) [reply tests/core/test_processor.py:1945]: It was indeed a duplicate curly brace, I missed it when I copied the response_selector dictionary from an example bot

**Pull request merged**